examples/benchmark_wattsstrogatz.py

```diff
@@ -0,0 +1,171 @@
+    offsets = list(range(1, fan_outs + 1))
+    offsets.extend([offs - n for offs in offsets])
+    if not directed:
+        offsets.extend([-offs for offs in offsets])
+    swmat = scipy_sparse.diags(np.ones((len(offsets), n), dtype=np.float32),
+                               offsets=offsets, shape=(n, n), format='csr')
+    # random rewires
+    n_edges = len(swmat.data)
+    rewires_bool = np.random.random(n_edges) < rewire_frac
+    rewire_inds = rewires_bool.nonzero()[0]
+    rewire_to = np.random.randint(n, size=rewire_inds.size)
+    swmat.indices[rewire_inds] = rewire_to
+    return swmat
+
+
+records = []
+
+for i, n_neurons in enumerate(ns_neurons):
+
+    rng = np.random.RandomState(123)
+    # a = rng.normal(scale=np.sqrt(1./dim), size=n_neurons)
+
+    # --- Model
+    with nengo.Network(seed=9) as model:
+        if sys.argv[1] == 'dl':
+            nengo_dl.configure_settings(inference_only=True)
+        # inputA = nengo.Node(a)
+        ens = nengo.Ensemble(n_neurons, 1)
+        # nengo.Connection(inputA, ens.neurons, synapse=0.03)
+
+        weimat = wattsstrogatz_adjacencies(n_neurons)
+        if sparse:
+            transform = nengo.transforms.Sparse((n_neurons, n_neurons), init=weimat)
+        else:
+            transform = weimat.toarray()
+        nengo.Connection(ens.neurons, ens.neurons, transform=transform, synapse=0.1)
+
+        # A_p = nengo.Probe(A.output, synapse=0.03)
+        E_p = nengo.Probe(ens.neurons, synapse=0.03)
+
+    # --- Simulation
+    try:
+        t_start = time.time()
+
+        # -- build
+        sim = sim_class(model, **sim_kwargs)
+        t_sim = time.time()
+
+        # -- warmup
+        sim.run(0.01)
+        t_warm = time.time()
+
+        # -- long-term timing
+        sim.run(simtime)
+        t_run = time.time()
+
+        # error for sanity checking (should be below ~0.25, definitely 0.5)
+        # not implemented
+
+        records.append(OrderedDict((
+            ('benchmark', 'watts-strogatz'),
+            ('name', sim_name),
+            ('n_neurons', n_neurons),
+            ('simtime', simtime),
+            ('status', 'ok'),
+            ('profiling', getattr(sim, 'profiling', 0)),
+            ('buildtime', t_sim - t_start),
+            ('warmtime', t_warm - t_sim),
+            ('runtime', t_run - t_warm),
+            )))
+        print(records[-1])
+        print("%s, n_neurons=%d successful" % (sim_name, n_neurons))
+        if getattr(sim, 'profiling', False):
+            sim.print_profiling(sort=1)
+    except Exception as e:
+        records.append(OrderedDict((
+            ('benchmark', 'watts-strogatz'),
+            ('name', sim_name),
+            ('n_neurons', n_neurons),
+            ('simtime', simtime),
+            ('status', 'exception'),
+            ('exception', str(e)),
+            )))
+        print(records[-1])
+        print("%s, n_neurons=%d exception" % (sim_name, n_neurons))
+        raise
+    finally:
+        try:
+            sim
```

Comment:
I'm guessing you copied this from another benchmark, which is out of date. We always use `with` blocks for the simulator now, which ensures that they close properly. So you can do something like `with sim_class(model, **sim_kwargs) as sim:` where the model is created, then you don't need this `finally` block at all.